jgrapht-core/src/main/java/org/jgrapht/graph/AbstractBaseGraph.java

```diff
@@ -39,32 +50,55 @@
  * @param <E> the graph edge type
  *
  * @author Barak Naveh
+ * @author Dimitrios Michail
  * @since Jul 24, 2003
  */
 public abstract class AbstractBaseGraph<V, E>
-    extends
-    AbstractGraph<V, E>
-    implements
-    Graph<V, E>,
-    Cloneable,
-    Serializable
+    extends AbstractGraph<V, E>
+    implements Graph<V, E>, Cloneable, Serializable
 {
     private static final long serialVersionUID = 4811000483921413364L;
 
     private static final String LOOPS_NOT_ALLOWED = "loops not allowed";
     private static final String GRAPH_SPECIFICS_MUST_NOT_BE_NULL =
         "Graph specifics must not be null";
 
-    private EdgeFactory<V, E> edgeFactory;
     private transient Set<V> unmodifiableVertexSet = null;
 
+    private Supplier<V> vertexSupplier;
+    private Supplier<E> edgeSupplier;
+    
+    @Deprecated
```

Comment:
Just a note that in the next version, when we remove this field, I think we'll need to break serialization compatibility at that time by updating the serialVersionUID (even though removing a field does not imply a serialVersionUID change in general).  I can't think of a way to avoid this unless we keep the EdgeFactory interface around for one additional version (since the only way to write a custom deserializer would be to wrap the old EdgeFactory after deserializing it.

Reply:
Right. I generated a new uid.